dbms/src/Functions/FunctionsHashing.h

```diff
@@ -707,6 +709,58 @@ class FunctionAnyHash : public IFunction
                 ErrorCodes::ILLEGAL_COLUMN);
     }
 
+    template <bool first>
+    void executeUUID(const IColumn * column, typename ColumnVector<ToType>::Container & vec_to)
+    {
+        if (const ColumnUInt128 * col_from = checkAndGetColumn<ColumnUInt128>(column))
+        {
+            const ColumnUInt128::Container & vec_from = col_from->getData();
+            size_t size = vec_from.size();
+            for (size_t i = 0; i < size; ++i)
+            {
+                std::string hash_string = toString(UUID(vec_from[i]));
+                const ToType h = Impl::apply(hash_string.c_str(), hash_string.size());
+                if (first)
+                    vec_to[i] = h;
+                else
+                    vec_to[i] = Impl::combineHashes(vec_to[i], h);
+            }
+        }
+        else
+        {
+            throw Exception("Illegal column " + column->getName()
+                            + " of argument of function " + getName(),
+                            ErrorCodes::ILLEGAL_COLUMN);
+        }
+    }
+
+    template <typename T, bool first>
+    void executeDecimal(const IColumn * column, typename ColumnVector<ToType>::Container & vec_to)
+    {
+        if (const ColumnDecimal<T> * col_from = checkAndGetColumn<ColumnDecimal<T>>(column))
+        {
+            const typename ColumnDecimal<T>::Container & vec_from = col_from->getData();
+            size_t size = vec_from.size();
+            for (size_t i = 0; i < size; ++i)
+            {
+                WriteBufferFromOwnString buf;
+                writeText(vec_from[i], vec_from.getScale(), buf);
```

Comment:
The same. Sorry for possible miscommunication.
We just need to apply hash function directly on byte representation in memory.